### PR TITLE
Ck/16683 correct focus on esc

### LINKED ISSUE #16683: Correctly return focus on ESC
There can be issues with focus management on ESC press when focus was in menu bar. Implement a way to prevent those issue from happening.

ref: https://github.com/ckeditor/ckeditor5/pull/15901#discussion_r1499392991
https://github.com/ckeditor/ckeditor5/blob/76919d55c7825bd4401d4a473d6415922f80b49e/packages/ckeditor5-ui/src/editorui/editorui.ts#L437-L464

### PR DESCRIPTION
**Pull request by mremiszewski**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Fix (ui): Correctly return focus on `Esc` after focusing the menu bar. Closes #16683.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

**Comment by DawidKossowski** [comment]
Please rebase this branch with `master`.

File: packages/ckeditor5-ui/src/editorui/editorui.ts [hunk 1]
```diff
@@ -393,18 +398,43 @@ export default abstract class EditorUI extends /* #__PURE__ */ ObservableMixin()
...
+			}
+
+			const focusedElement = this.focusTracker.focusedElement as HTMLElement;
+
+			// If focus moved out of a DOM element that does not belong to the editing view (e.g. source editing).
+			if (
+				Array.from( this._editableElementsMap.values() ).includes( focusedElement ) &&
+				!Array.from( this.editor.editing.view.domRoots.values() ).includes( focusedElement )
+			) {
+				this._lastFocusedForeignElement = focusedElement;
```
Comment (oleq) [inline packages/ckeditor5-ui/src/editorui/editorui.ts:432]: Could we extract this to a common (for alt+10 and alt+f9 listeners) private method like `_updateLastFocusedElement()`? It looks like this bit repeats in both listeners. WDYT?

Comment (mremiszewski) [reply packages/ckeditor5-ui/src/editorui/editorui.ts:432]: Okay, I can do that

**Comment by DawidKossowski** [comment]
@mremiszewski please correct the commit history.

**Review by oleq — APPROVED** [review]

**Pull request merged**